wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -115,38 +127,31 @@ class CoinJoinMixingService @Inject constructor(
     private var mixingCompleteListeners: ArrayList<MixingCompleteListener> = arrayListOf()
     private var sessionCompleteListeners: ArrayList<SessionCompleteListener> = arrayListOf()
 
-    override var mode: CoinJoinMode = CoinJoinMode.BASIC
+    var mode: CoinJoinMode = CoinJoinMode.NONE
     override var mixingStatus: MixingStatus = MixingStatus.NOT_STARTED
         private set
 
     private val coroutineScope = CoroutineScope(
-        Executors.newFixedThreadPool(2).asCoroutineDispatcher(),
+        Executors.newFixedThreadPool(2, ContextPropagatingThreadFactory("coinjoin-pool")).asCoroutineDispatcher()
```

Comment:
`ContextPropagatingThreadFactory` will call `Context.propogate` in the created threads and we don't need to worry about missing Context exceptions.

Reply:
That's neat.